test/validation.html

```diff
@@ -334,7 +334,40 @@
             expect(field.invalid).to.be.true;
           });
         });
+      });
+    });
+
+    describe('initial validation', () => {
```

Comment:
Let's create separate files per component – `text-field-validation.html`, `number-field-validation.html`, etc – with the following structure:

```js
describe('text-field validation', () => {
  describe('initial', () => {
    // add tests related to the initial validation here
  });

  describe('basic', () => {
    // add the validated event tests here
  });

  describe('invalid cannot be set to false', () => {
    // add tests related to shouldSetInvalid here
  });
});
```

Let's try that approach in this PR and then update other PRs after this one is merged.